e2e/testcases/admission_test.go

```diff
@@ -141,55 +148,84 @@ metadata:
-metadata:
-  annotations:
-    configsync.gke.io/webhook-configuration-update: disabled
-  name: admission-webhook.configsync.gke.io
-  labels:
-    configmanagement.gke.io/system: "true"
-    configmanagement.gke.io/arch: "csmr"
-`)
-
 	nt := nomostest.New(t, nomostesting.DriftControl)
 
+	// Test starts with Admission Webhook already installed
+	nomostest.WaitForWebhookReadiness(nt)
+
 	nt.Must(nt.RootRepos[configsync.RootSyncName].Add("acme/namespaces/hello/ns.yaml", fake.NamespaceObject("hello")))
 	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("add Namespace"))
 	if err := nt.WatchForAllSyncs(); err != nil {
 		nt.T.Fatal(err)
 	}
 
-	if err := os.WriteFile(filepath.Join(nt.TmpDir, "webhook.yaml"), webhook, 0644); err != nil {
-		nt.T.Fatalf("failed to create a tmp file %v", err)
+	helloNS := &corev1.Namespace{}
+	if err := nt.KubeClient.Get("hello", "", helloNS); err != nil {
+		nt.T.Fatal(err)
 	}
+	version := helloNS.ResourceVersion
 
-	// Recreate the admission webhook
-	if _, err := nt.Shell.Kubectl("replace", "-f", filepath.Join(nt.TmpDir, "webhook.yaml")); err != nil {
-		nt.T.Fatalf("failed to replace the admission webhook %v", err)
-	}
+	nomostest.StopWebhook(nt)
 
-	// Verify that the webhook is disabled.
-	if _, err := nt.Shell.Kubectl("delete", "ns", "hello"); err != nil {
-		nt.T.Fatalf("failed to run `kubectl delete ns hello` %v", err)
+	tg := taskgroup.New()
+	tg.Go(func() error {
+		return nt.Validate(
+			core.RootReconcilerName(configsync.RootSyncName),
+			configsync.ControllerNamespace,
+			&appsv1.Deployment{},
+			testpredicates.DeploymentMissingEnvVar(reconcilermanager.Reconciler, reconcilermanager.WebhookEnabled),
+		)
+	})
+	if err := tg.Wait(); err != nil {
+		nt.T.Fatal(err)
 	}
-
-	// Remove the annotation for disabling webhook
-	if _, err := nt.Shell.Kubectl("annotate", "ValidatingWebhookConfiguration", "admission-webhook.configsync.gke.io", "configsync.gke.io/webhook-configuration-update-"); err != nil {
-		nt.T.Fatalf("failed to remove the annotation in the admission webhook %v", err)
+	nt.T.Logf("Check if the namespace resourceversion has changed after deleting Admission Webhook")
+	err := nt.Watcher.WatchObject(kinds.Namespace(), "hello", "",
+		[]testpredicates.Predicate{
+			testpredicates.ResourceVersionNotEquals(nt.Scheme, version),
+		})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	nt.T.Logf("Check if the test namespace has dropped the declared-field annotation")
+	if err := nt.Validate("hello", "", &corev1.Namespace{}, testpredicates.MissingAnnotation(metadata.DeclaredFieldsKey)); err != nil {
+		nt.T.Fatal(err)
 	}
 
-	nt.Must(nt.RootRepos[configsync.RootSyncName].Add("acme/namespaces/test/ns.yaml", fake.NamespaceObject("test")))
-	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("add another Namespace"))
-	if err := nt.WatchForAllSyncs(); err != nil {
+	helloNS = &corev1.Namespace{}
+	if err := nt.KubeClient.Get("hello", "", helloNS); err != nil {
```

Comment:
The Get request is to get the resourceVersion. I don't think we care about the resourceVersion that much. I think validating the update of the declared-fields annotation should be good enough.

Reply:
Adding the annotation check as another task would fail straight away, I'm not sure if `watch` fits better for this case, but waiting for the new root-reconciler to come up -> check test object updated -> check annotation is gone seems necessary

Reply:
Resolved in earlier comment.